### ISSUE TITLE
Support changing permissions on secret directories

### ISSUE THREAD

**Issue by virchau13** [Mic92/sops-nix#150]
I have a whole bunch of secrets stored in the `/run/secrets/scripts` directory. I want to put the directory `/run/secrets/scripts` into a non-root user's $PATH, but since that user doesn't have permission to read the directory, it doesn't work. 

This can be worked around by manually symlinking to the secrets, but it would be great to have this as a normal feature.

**Comment by Mic92** [comment]
Not sure what the configuration would look for this since permissions are nested...

**Comment by dasJ** [comment]
Two ideas pop into my head, one in yaml, one in nix:
```yaml
slapd:
  __dirMode: 0755
  __dirOwner: slapd
  password: abc
```

```nix
{
  sops.secrets."slapd/password" = {};
  sops.dirPermissions.slapd = {
    mode = "0755";
    owner = "slapd";
  };
}
```

**Comment by Mic92** [comment]
Mhm, this seems like quite a bit of extra complexity for each feature that not many people use.

**Comment by sector1209** [comment]
> > Mhm, this seems like quite a bit of extra complexity for each feature that not many people use.
> 
> I use sops-nix to put secrets into the .ssh directory of my main User. I just installed a new system and sops-nix created .ssh directory automatically because that's where my secrets are supposed to go, would be perfect if only .ssh wasn't owned by root and ssh could change known-hosts in the .ssh directory. This is easily resolved with `sudo chown username .ssh` but not a solutioin considering that we all switched to nix for declarative configuration.

I'm also struggling with this issue currently :/ did you ever find a solution?

**Comment by jaredmontoya** [comment]
No, I have to manually `chown` the created parent directory to the proper user.
